plugins/inputs/tail/tail.go

```diff
@@ -205,7 +228,20 @@ func (t *Tail) Stop() {
 	t.Lock()
 	defer t.Unlock()
 
+	offsetsMutex.Lock()
+	defer offsetsMutex.Unlock()
+
 	for _, tailer := range t.tailers {
+		if !t.Pipe && !t.FromBeginning {
+			// store offset for resume
+			offset, err := tailer.Tell()
```

Comment:
Is it safe to call this after the tailer has Stopped?

Reply:
Ideally, but if you stop first Tell returns an offset of 0.

when tailFileSync() finishes, it calls tail.close() which calls tail.closeFile() which sets tailer.file = nil, which causes Tell() to return an zero value.

Reply:
Slightly unfortunate, but worse case we reprocess some lines.

Reply:
An improvement to the tail lib could resolve this